plugins/modules/source_control/gitlab/gitlab_group.py

```diff
@@ -212,6 +218,10 @@ def create_or_update_group(self, name, parent, options):
             if options.get('require_two_factor_authentication'):
                 payload['require_two_factor_authentication'] = options['require_two_factor_authentication']
             group = self.create_group(payload)
+
+            # add avatar to group
+            if options['avatar_path']:
+                group.avatar = open(options['avatar_path'], 'rb')
```

Comment:
You need some logic to handle the case that the file does not exist. How about:
```suggestion
                try:
                    group.avatar = open(options['avatar_path'], 'rb')
                except IOError as e:
                    self._module.fail_json(msg='Cannot open {0}: {1}'.format(options['avatar_path'], e))
```